app/mailers/kudo_mailer.rb

```diff
@@ -10,7 +10,11 @@ class KudoMailer < ActionMailer::Base
   #   guest_count: number of guest kudos
   def batch_kudo_notification(user_id, user_kudos)
     @commentables = []
+    @kudo_counts = {}
     @kudo_givers = {}
+    # Does this work have only one kudos, and is it from a guest? We need this
+    # information in the view.
+    @single_guest_only = {}
```

Comment:
I will gladly take suggestions for a better name.

Reply:
Oh, also: I considered making an array containing the commentables that matched this criteria and then, in the view, checking if the current commentable was in the array to determine which translation we wanted to use. It doesn't really save any code and is just as difficult to name, but if it seems like it would be clearer, I can switch to that.

Reply:
I think we can have the mailer do one pass through `kudos_hash` to build a list of commentables that exist. The template gets `@commentables` and `user_kudos` (hash from `[commentable_type]_[commentable_id]` to: list of registered user names, a guest count) and decides which translation to pick from there.

So most of the logic will be in the templates.